### PR TITLE
Added the reducers and moved the store

### PR DESCRIPTION
**Pull request by Pdzly**
Is the 2nd part of #66 changes the reducers to use the redux-toolkit slices.

### CODE REVIEW

File: src/redux/reducer/configReducer.js [hunk 1]
```diff
@@ -56,30 +43,36 @@ const initialState = {
...
   }
-};
+})
+
+
+
+export default configReducer.reducer;
 
-export default configReducer;
+export const { setConfigItem, setConfigItemJson } = configReducer.actions;
```
Comment (tgxn) [inline src/redux/reducer/configReducer.js:75]: i like this approach 👍

File: src/redux/store.js
Comment (tgxn) [inline src/redux/store.js:1]: does this basically store and hydrate the redux store to `localStorage` or something?

could be good if we could remove the JSON storage in `configReducer`?

File: src/redux/store.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+} from "redux-persist";
+import thunk from "redux-thunk";
+import { setupListeners } from "@reduxjs/toolkit/dist/query";
+
+import accountReducer from "./reducer/accountReducer";
+import configReducer from "./reducer/configReducer";
+
+
+export const store = configureStore({
+  reducer: {
```
Comment (tgxn) [inline src/redux/store.js:19]: import these as from `src/reducers.js` ?

File: src/redux/reducer/configReducer.js [hunk 3]
```diff
@@ -56,30 +43,36 @@ const initialState = {
   nsfwWords: loadWithDefault("config.nsfwWords", []),
 };
 
-const configReducer = (state = initialState, action = {}) => {
-  switch (action.type) {
-    case "setConfigItem":
+
+const configReducer = createSlice({
```
Comment (tgxn) [inline src/redux/reducer/configReducer.js:47]: this usage is far cleaner, does this basically mean that `createSlice` creates the actions for us based on the reducers? 😲

**Review by tgxn — COMMENTED** [review]
looks nice, great work with `@reduxjs/toolkit` :D

File: src/redux/store.js
Comment (Pdzly) [reply src/redux/store.js:1]: It isnt persisting it in the local storage. But i would keep it like this, it is a bit more clearer where it is set.

Comment (tgxn) [reply src/redux/store.js:1]: what is the benefit of using `redux-persist` in this way?

File: src/redux/store.js (hunk 2 above)
Comment (Pdzly) [reply src/redux/store.js:19]: Yeah i oversaw that those are in reducers.js but normally its named "store" ( as it is the root for the createAPI from toolkit if we want to use it in the future )

File: src/redux/reducer/configReducer.js (hunk 3 above)
Comment (Pdzly) [reply src/redux/reducer/configReducer.js:47]: Yes

File: src/redux/reducer/configReducer.js (hunk 1 above)
Comment (Pdzly) [reply src/redux/reducer/configReducer.js:75]: Glad it looks good for you :)

**Comment by tgxn** [comment]
nice work 👍 in future, can we merge stuff into `develop` first - so it will build a container, and we can bundle releases to main 😎
im gonna merge my pm stuff now so i wanna check if this all works with that

**Review by tgxn — APPROVED** [review]

**Comment by Pdzly** [comment]
Oh yeah i am sry i forgot it agiain @tgxn ! I will remind myself next time.

**Comment by tgxn** [comment]
hmmmmmm, looks like there's some issues with this though. I'm getting some errors on develop :/ `currentUser.site is undefined`
seeing an undefined `site`
image

**Comment by tgxn** [comment]
https://github.com/tgxn/lemmy-modder/compare/develop...fix_reducers?expand=1
fixes in here should be OK

**Pull request merged**